coordinator/app/src/main/kotlin/net/consensys/zkevm/ethereum/coordination/SubmittedEventTracker.kt

```diff
@@ -0,0 +1,36 @@
+package net.consensys.zkevm.ethereum.coordination
+
+import net.consensys.zkevm.domain.BlobSubmittedEvent
+import net.consensys.zkevm.domain.FinalizationSubmittedEvent
+import tech.pegasys.teku.infrastructure.async.SafeFuture
+import java.util.concurrent.atomic.AtomicReference
+import java.util.function.Supplier
+
+abstract class SubmittedEventTracker<T>(submission: T?) : (T?) -> SafeFuture<*>, Supplier<T?> {
```

Comment:
I think we don't leverage `, Supplier<T?>` part anywhere

Reply:
And I don't see where we actually access the whole stored submission. In the two implementations we have, we only access the ULong, it means we don't need to store the whole T. And this means that existing `MaxLongTracker` should be sufficient